tests/utils.py

```diff
@@ -137,6 +138,36 @@ def generate_core_file(
             process.wait(timeout=TIMEOUT)
 
 
+@contextlib.contextmanager
+def generate_core_at_breakpoint(
```

Comment:
This function's name no longer matches what it does, since you don't actually set any breakpoints. `generate_core_after_crash` or something would be more appropriate.